ql/src/java/org/apache/hadoop/hive/ql/metadata/Hive.java

```diff
@@ -4475,7 +4481,8 @@ public static boolean moveFile(final HiveConf conf, Path srcf, final Path destf,
         destFs.copyFromLocalFile(srcf, destf);
         return true;
       } else {
-        if (needToCopy(conf, srcf, destf, srcFs, destFs, configuredOwner, isManaged)) {
+        if (needToCopy(conf, srcf, destf, srcFs, destFs, configuredOwner, isManaged, isCompactionTable,
+            isMmCompactionTable)) {
           //copy if across file system or encryption zones.
           LOG.debug("Copying source " + srcf + " to " + destf + " because HDFS encryption zones are different.");
```

Comment:
Would it make sense to include the hashes for the fs instances in the debug logs?

Reply:
I think only in hdfs-common (FS.close() et al) – I opened HADOOP-17267 to track it.